### PR TITLE
Handle VecOfArray

### PR DESCRIPTION
**Pull request by dhairyagandhi96**
With sister PR https://github.com/SciML/RecursiveArrayTools.jl/pull/86

Checking to see what CI says

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
I am not sure when that branch is hit? Do we have an example?

**Comment by dhairyagandhi96** [comment]
Happened for https://github.com/SciML/DiffEqFlux.jl/issues/160

**Comment by dhairyagandhi96** [comment]
Not happy about the branching here

**Comment by ChrisRackauckas** [comment]
It doesn't look like this computes correct values. Added a test.

**Comment by dhairyagandhi96** [comment]
Looking into the failure, hopefully it's just in the adjoint

**Comment by ChrisRackauckas** [comment]
I think using `sum` is incorrect: it should just be the identity.

File: src/local_sensitivity/concrete_solve.jl [hunk 1]
```diff
@@ -65,7 +65,11 @@ function DiffEqBase._concrete_solve_adjoint(prob,alg,
       if only_end
         _out[:] .= -vec(Δ)
       else
-        _out[:] .= -adapt(typeof(u0),reshape(Δ, prod(size(Δ)[1:end-1]), size(Δ)[end])[:, i])
+        if typeof(Δ) <: AbstractArray{<:AbstractArray}
+          _out[:] .= sum.(Δ)
```
Comment (ChrisRackauckas) [inline src/local_sensitivity/concrete_solve.jl:69]: Maybe `_out[:] .= Δ[i]`

Comment (dhairyagandhi96) [reply src/local_sensitivity/concrete_solve.jl:69]: I'll try that out

**Comment by dhairyagandhi96** [comment]
CI is green now

**Pull request merged**